platform/darwin/docs/guides/Predicates and Expressions.md

```diff
@@ -19,6 +19,17 @@ based on the feature’s attributes. Use the `MGLVectorStyleLayer.predicate`
 property to include only the features in the source layer that satisfy a
 condition that you define.
 
+When a predicate is used for filtering vector data the key should be cast
+explicitly into the key's type. A number based key will become: `CAST(key, 'NSNumber')`
+
+`NSPredicateOperatorType`                     | Format string syntax
+----------------------------------------------|---------------------
+`NSGreaterThanOrEqualToPredicateOperatorType` | `CAST(key, 'valueType') >= value`<br />`CAST(key, 'valueType') => value`
+`NSLessThanOrEqualToPredicateOperatorType`    | `CAST(key, 'valueType') <= value`<br />`CAST(key, 'valueType') =< value`
+`NSGreaterThanPredicateOperatorType`          | `CAST(key, 'valueType') > value`
+`NSLessThanPredicateOperatorType`             | `CAST(key, 'valueType') < value`
```

Comment:
This table is redundant to the table below it, which introduces these operators. Instead, add a paragraph below that table mentioning that `key` may need to be cast to a number or string using the `CAST()` operator and the type `NSNumber` or `NSString`. (Mention these types explicitly, because Swift developers may be inclined to use `Float` or `String` instead.)